### DISCUSSION TITLE
Is formio js  the opensource version stable enough to use in a project

### DISCUSSION THREAD

**Discussion by masudchowdhuryca** [formio/formio.js#4482] [General]
Is formio js  the opensource version stable enough to use in a project ?
will i face any issue of crash or data loss in future ?

**Comment by AnthonyPhan** [comment]
I cannot speak to the backend but I have defiantly used the form render in production and is working well.

**Comment by 0liver** [comment]
The answer is probably: It depends.

It depends on the complexity of the forms you are planning to build. See the [currently open issues](https://github.com/formio/formio.js/issues) to get an impression of situations that are not yet ironed out.
It depends on your ability and willingness to dive into the source code and find out why some things don't work as expected.
It depends on your expectations. I haven't had any "crashes" (whatever that means in the context of a JS library), but I've read of accounts of data loss in some complicated examples. I haven't experienced any data loss with the self-hosted version myself, though.
